# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
# Commit: 675b46e5aba7
# Full Hash: 675b46e5aba7aab81b45a56a8ea1270ae657514a
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 3
# Description:
#   Bug 1654112 - Destroy Call instances as their associated PeerConnections close. r=ng,bwc
#   
#   Prior to this patch, WebRtcCallWrapper would never clear the call instance,
#   meaning that it went away as the classes having references to WebRtcCallWrapper
#   also went away. These are mainly TransceiverImpl and PeerConnectionMedia. There
# ==============================================================================

diff -r da26bac32c32 -r 675b46e5aba7 dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Tue Jan 19 10:59:28 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Thu Jan 28 11:15:40 2021 +0100
@@ -965,6 +965,10 @@
   auto syncPromise =
       InvokeAsync(GetMainThreadSerialEventTarget(), __func__, [&] {
         auto current = TaskQueueWrapper::MainAsCurrent();
+        if (!mCall->Call()) {
+          return PacketPromise::CreateAndResolve(
+              webrtc::PacketReceiver::DELIVERY_PACKET_ERROR, __func__);
+        }
         // Bug 1499796 - we need to get passed the time the
         // packet was received
         webrtc::PacketReceiver::DeliveryStatus status =